quickjs-c-atomics.h

```diff
@@ -49,6 +49,47 @@
 #define atomic_fetch_sub(obj, arg) \
     __atomic_fetch_sub(obj, arg, __ATOMIC_SEQ_CST)
 #define _Atomic
+#else
+#if defined(_MSC_VER) && _MSC_VER < 1935
+#define _Atomic
+/*
+ * Copypasted from: https://github.com/videolan/dav1d/blob/master/include/compat/msvc/stdatomic.h
+ */
+#include <windows.h>
+
+typedef volatile LONG  atomic_int;
+typedef volatile ULONG atomic_uint;
+
+typedef enum {
+    memory_order_relaxed,
+    memory_order_acquire
+} msvc_atomic_memory_order;
+
+#define atomic_init(p_a, v)           do { *(p_a) = (v); } while(0)
+#define atomic_store(p_a, v)          InterlockedExchange((LONG*)p_a, v)
+#define atomic_load(p_a)              InterlockedCompareExchange((LONG*)p_a, 0, 0)
+#define atomic_exchange(p_a, v)       InterlockedExchange(p_a, v)
+#define atomic_load_explicit(p_a, mo) atomic_load(p_a)
+
+static inline int atomic_compare_exchange_strong_int(LONG *obj, LONG *expected,
+                                                     LONG desired)
+{
+    LONG orig = *expected;
+    *expected = InterlockedCompareExchange(obj, desired, orig);
+    return *expected == orig;
+}
+#define atomic_compare_exchange_strong(p_a, expected, desired) atomic_compare_exchange_strong_int((LONG *)p_a, (LONG *)expected, (LONG)desired)
```

Comment:
This and atomic_load/atomic_store/etc. aren't safe. quickjs.c definitely calls them on values that aren't sizeof(LONG).

Reply:
Will removing the casting to `LONG` make them safe?

Reply:
No. There's https://learn.microsoft.com/en-us/previous-versions/ttk2z1ws(v=vs.85) for compare-and-swap for 16, 32 and 64 bits values, but not 8 bits, AFAIK.

Reply:
Will replacing these functions with `_` alternatives do?

Reply:
My suggestion would be to use [`_Generic`](https://en.cppreference.com/w/c/language/generic), to dispatch based on the type. That _should_ (fingers crossed) work with 2019.